Looks like it is failing now for some reason. Weirdly, the error message points to a file that does not contain the include directive the message says.

```
[64/86] Compiling C object test/polkit/polkitidentitytest.p/polkitidentitytest.c.o-agent-1.so.0.0.0.symbols
FAILED: test/polkit/polkitidentitytest.p/polkitidentitytest.c.o 
gcc -Itest/polkit/polkitidentitytest.p -Itest/polkit -I../test/polkit -Isrc -I../src -Isrc/polkit -I/nix/store/150bwrv7p7qhb0vm0zwkw9z980gc58nq-glib-2.70.1-dev/include -I/nix/store/150bwrv7p7qhb0vm0zwkw9z980gc58nq-glib-2.70.1-dev/include/glib-2.0 -I/nix/store/5dckamm3ls6mgyp4p0nd8dvpv499m1a4-glib-2.70.1/lib/glib-2.0/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -DHAVE_CONFIG_H -D_POLKIT_COMPILATION -D_POLKIT_BACKEND_COMPILATION -MD -MQ test/polkit/polkitidentitytest.p/polkitidentitytest.c.o -MF test/polkit/polkitidentitytest.p/polkitidentitytest.c.o.d -o test/polkit/polkitidentitytest.p/polkitidentitytest.c.o -c ../test/polkit/polkitidentitytest.c
../test/polkit/polkitidentitytest.c:22:10: fatal error: config.h: No such file or directory
   22 | #include "config.h"
      |          ^~~~~~~~~~
compilation terminated.
[65/86] Compiling C object test/polkit/polkitunixnetgrouptest.p/polkitunixnetgrouptest.c.o
FAILED: test/polkit/polkitunixnetgrouptest.p/polkitunixnetgrouptest.c.o 
gcc -Itest/polkit/polkitunixnetgrouptest.p -Itest/polkit -I../test/polkit -Isrc -I../src -Isrc/polkit -I/nix/store/150bwrv7p7qhb0vm0zwkw9z980gc58nq-glib-2.70.1-dev/include -I/nix/store/150bwrv7p7qhb0vm0zwkw9z980gc58nq-glib-2.70.1-dev/include/glib-2.0 -I/nix/store/5dckamm3ls6mgyp4p0nd8dvpv499m1a4-glib-2.70.1/lib/glib-2.0/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -DHAVE_CONFIG_H -D_POLKIT_COMPILATION -D_POLKIT_BACKEND_COMPILATION -MD -MQ test/polkit/polkitunixnetgrouptest.p/polkitunixnetgrouptest.c.o -MF test/polkit/polkitunixnetgrouptest.p/polkitunixnetgrouptest.c.o.d -o test/polkit/polkitunixnetgrouptest.p/polkitunixnetgrouptest.c.o -c ../test/polkit/polkitunixnetgrouptest.c
../test/polkit/polkitunixnetgrouptest.c:22:10: fatal error: config.h: No such file or directory
   22 | #include "config.h"
      |          ^~~~~~~~~~
compilation terminated.
```

--- a/pkgs/development/libraries/polkit/default.nix
+++ b/pkgs/development/libraries/polkit/default.nix
@@ -1,11 +1,15 @@
 { lib
 , stdenv
-, fetchurl
+, fetchFromGitLab
 , pkg-config
 , glib
 , expat
 , pam
+, meson
+, ninja
 , perl
+, rsync
+, python3
 , fetchpatch
 , gettext
 , spidermonkey_78
@@ -38,12 +42,16 @@ stdenv.mkDerivation rec {
 
   outputs = [ "bin" "dev" "out" ]; # small man pages in $bin
 
-  src = fetchurl {
-    url = "https://www.freedesktop.org/software/${pname}/releases/${pname}-${version}.tar.gz";
-    sha256 = "7npZmoUxF78nNUhyVxn6kvq9LxNpFcekkGzumFZ67gM=";
+  # Tarballs do not contain subprojects.
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "polkit";
+    repo = "polkit";
+    rev = version;
+    sha256 = "oEaRf1g13zKMD+cP1iwIA6jaCDwvNfGy2i8xY8vuVSo=";
   };
 
-  patches = [
+  patches = lib.optionals stdenv.hostPlatform.isMusl [
     # Make netgroup support optional (musl does not have it)
     # Upstream MR: https://gitlab.freedesktop.org/polkit/polkit/merge_requests/10
     # We use the version of the patch that Alpine uses successfully.
@@ -59,7 +67,17 @@ stdenv.mkDerivation rec {
     gtk-doc
     pkg-config
     gettext
+    meson
+    ninja
     perl
+    rsync
+    (python3.withPackages (pp: with pp; [
+      dbus-python
+      (python-dbusmock.overridePythonAttrs (attrs: {
+        # Avoid dependency cycle.
+        doCheck = false;
+      }))
+    ]))
 
     # man pages
     libxslt
@@ -86,36 +104,33 @@ stdenv.mkDerivation rec {
     dbus
   ];
 
-  configureFlags = [
+  mesonFlags = [
     "--datadir=${system}/share"
     "--sysconfdir=/etc"
-    "--with-systemdsystemunitdir=${placeholder "out"}/etc/systemd/system"
-    "--with-polkitd-user=polkituser" #TODO? <nixos> config.ids.uids.polkituser
-    "--with-os-type=NixOS" # not recognized but prevents impurities on non-NixOS
-    (if withIntrospection then "--enable-introspection" else "--disable-introspection")
-  ] ++ lib.optionals (!doCheck) [
-    "--disable-test"
+    "-Dsystemdsystemunitdir=${placeholder "out"}/etc/systemd/system"
+    "-Dpolkitd_user=polkituser" #TODO? <nixos> config.ids.uids.polkituser
+    "-Dos_type=redhat" # only affects PAM includes
+    "-Dintrospection=${lib.boolToString withIntrospection}"
+    "-Dtests=${lib.boolToString doCheck}"
+    "-Dgtk_doc=${lib.boolToString true}"
+    "-Dman=true"
+  ] ++ lib.optionals stdenv.isLinux [
+    "-Dsession_tracking=${if useSystemd then "libsystemd-login" else "libelogind"}"
   ];
 
-  makeFlags = [
-    "INTROSPECTION_GIRDIR=${placeholder "out"}/share/gir-1.0"
-    "INTROSPECTION_TYPELIBDIR=${placeholder "out"}/lib/girepository-1.0"
-  ];
-
-  installFlags = [
-    "datadir=${placeholder "out"}/share"
-    "sysconfdir=${placeholder "out"}/etc"
-  ];
+  # HACK: We want to install policy files files to $out/share but polkit
+  # should read them from /run/current-system/sw/share on a NixOS system.
+  # Similarly for config files in /etc.
+  # With autotools, it was possible to override Make variables
+  # at install time but Meson does not support this
+  # so we need to convince it to install all files to a temporary
+  # location using DESTDIR and then move it to proper one in postInstall.
+  DESTDIR = "${placeholder "out"}/dest";
 
   inherit doCheck;
 
-  postPatch = lib.optionalString stdenv.isDarwin ''
-    sed -i -e "s/-Wl,--as-needed//" configure.ac
-  '';
-
-  preConfigure = ''
-    chmod +x test/mocklibc/bin/mocklibc{,-test}.in
-    patchShebangs .
+  postPatch = ''
+    patchShebangs test/polkitbackend/polkitbackendjsauthoritytest-wrapper.py
 
     # ‘libpolkit-agent-1.so’ should call the setuid wrapper on
     # NixOS.  Hard-coding the path is kinda ugly.  Maybe we can just
@@ -125,22 +140,38 @@ stdenv.mkDerivation rec {
     substituteInPlace test/data/etc/polkit-1/rules.d/10-testing.rules \
       --replace   /bin/true ${coreutils}/bin/true \
       --replace   /bin/false ${coreutils}/bin/false
+  '';
 
-  '' + lib.optionalString useSystemd /* bogus chroot detection */ ''
-    sed '/libsystemd autoconfigured/s/.*/:/' -i configure
+  postConfigure = ''
+    # Unpacked by meson
+    chmod +x subprojects/mocklibc-1.0/bin/mocklibc
+    patchShebangs subprojects/mocklibc-1.0/bin/mocklibc
   '';
 
   checkPhase = ''
     runHook preCheck
 
-    # unfortunately this test needs python-dbusmock, but python-dbusmock needs polkit,
-    # leading to a circular dependency
-    substituteInPlace test/Makefile --replace polkitbackend ""
-
     # tests need access to the system bus
-    dbus-run-session --config-file=${./system_bus.conf} -- sh -c 'DBUS_SYSTEM_BUS_ADDRESS=$DBUS_SESSION_BUS_ADDRESS make check'
+    dbus-run-session --config-file=${./system_bus.conf} -- sh -c 'DBUS_SYSTEM_BUS_ADDRESS=$DBUS_SESSION_BUS_ADDRESS meson test --print-errorlogs'
 
     runHook postCheck
+  '';
+
+  postInstall = ''
+    # Move stuff from DESTDIR to proper location.
+    # We use rsync to merge the directories.
+    rsync --archive "${DESTDIR}/etc" "$out"
+    rm --recursive "${DESTDIR}/etc"
+    rsync --archive "${DESTDIR}${system}"/* "$out"
+    rm --recursive "${DESTDIR}${system}"/*
+    rmdir --parents --ignore-fail-on-non-empty "${DESTDIR}${system}"
+    for o in $outputs; do
+        rsync --archive "${DESTDIR}/''${!o}" "$(dirname "''${!o}")"
+        rm --recursive "${DESTDIR}/''${!o}"
+    done
+    # Ensure the DESTDIR is removed.
+    destdirContainer="$(dirname "${DESTDIR}")"
+    pushd "$destdirContainer"; rmdir --parents "''${DESTDIR##$destdirContainer/}${builtins.storeDir}"; popd
   '';
 
   meta = with lib; {

